### DISCUSSION TITLE
新しい Vue.js ドキュメントの翻訳について

### DISCUSSION THREAD

**Discussion by kazupon** [vuejs-translations/docs-ja#2] [General]
@jay-es @naokie @kiaking 

[Vue.js公式ブログ](https://blog.vuejs.org/posts/vue-3-as-the-new-default.html)のアナウンスでもありましたが、2/7 に npm でインストール可能な Vue.js のメインストリームがいよいよ Vue 3 に切り替わります。
それに伴い、Vue.js 公式ドキュメントも [staging.vuejs.org](https://staging.vuejs.org/) でホストされているものが、vuejs.org に変わる予定です。

staging.vuejs.org でホストされているドキュメントは、[v3.vuejs.org](https://v3.vuejs.org)をベースにre-fineされたものですが、Evan 氏によってほぼ全面書き換えられています。
ですので、既存のv3.vuejs.orgから流用しながらの翻訳は今回は厳しそうです。(もしかしたら、理由用できるものがあるかも)
一から翻訳した方がよいと考えています。

## ドキュメントの構成
新しいドキュメントの構成ですが、ディレクトリ構成は以下のようになっています。
https://github.com/vuejs/docs/tree/main/src

新しいドキュメントは VitePress で markdown がビルドされホストされます。

以下は staging.vuejs.org に表示されるナビゲーション部分をキャプチャしたものです。
![image](https://user-images.githubusercontent.com/72989/151391505-6eca7b98-9267-4590-96dc-27aba47c11fd.png)

VitePress で構築されるドキュメントは、v3.vuejs.org のときと同じく、メニューとディレクトリで構成されています。
そして、ディレクトリには、章単位で makrdown ファイルが存在する形となっています。

## 翻訳方法
翻訳ですが、これまでの Vue.js ドキュメント翻訳でやってきたやり方でやるといいと思っています。
ナビゲーションのメニュー単位でディレクトリがあり、その中に makrdown ファイルがあるので、1 markdown を一人が翻訳するという形です。

## 翻訳状況の管理方法
翻訳状況の管理については、GitHub issues を使うとよいでしょう。
ナビゲーションのメニュー単位、つまりディレクトリ単位 (e.g. メニュー: ガイド、ディレクトリ `src/guide`) で issue を作り、その issue に メニューに対応するディレクトリにある markdown ファイルですべてチェックボックス付きでリスト化します。

そして、1人 1 markdown ファイルを翻訳作業できるように、章タイトル名付きで GitHub issue 化します。
その issue 化したものを管理できるように、issue 番号で対象となるメニューの issue に紐付けます。

markdown ファイルですべてチェックボックスによるリスト化、そして 1 markdown 1 issue については、 @naokie さんが対応されたこちらが参考になります。
https://github.com/vuejs-jp/ja.vuejs.org/issues/347

## 翻訳者募集方法
v3.vuejs.org の翻訳と同じく、vue.js-jp の slack アナウンスして募集します。
また、翻訳開始したら、vuejs-jp の Twitter アカウントでもアナウンスします。

## 翻訳途中の本家ドキュメントのアップデートの差分
@kiaking さんが作った [ryo-cho](https://github.com/vuejs-translations/ryu-cho) を使って GitHub Actions によって差分を issue 化します。

## ホスト先
vue.js が持っている netlify を使ってホストします。
ドメインの割当は、Evan 氏もしくは、Ben 氏に対応して貰う予定です。

## 参考
すでに中国の方では、翻訳が進んでいます。
https://github.com/vuejs-translations/docs-zh-cn

**Comment by kazupon** [comment]
@jay-es @naokie @kiaking 
こんな感じですが、どうでしょう？
私の方で、翻訳方針について書きましたが、何かいいアイデア、もっとこうした方がいいよ！っていうのがありました、コメントください。 🙏

**Reply by naokie** [reply]
ありがとうございます！自分はほぼ合意です 👍🏼 

### 流用できそうか

https://github.com/vuejs/docs/blob/main/src/guide/essentials/computed.md
https://github.com/vuejs/docs/blob/master/src/guide/computed.md
1文1文を見てみると、流用できるところもありますが、それを探すほうが大変だったりして、
翻訳はやり直したほうがよさそうとは考えていました。

### 翻訳方法

担当の割り振り方は、マークダウンのファイル単位でやったほうが、
1ページ内に繰り返される単語のブレが減るので、妥当な粒度だと以前のドキュメント翻訳をやって思いました。

### 加えるなら

textlint と[用語集](https://github.com/vuejs-jp/ja.vuejs.org/wiki/%E7%94%A8%E5%AD%97%E3%80%81%E7%94%A8%E8%AA%9E) も引き続き、活用できるようにしていきたいです。


中国語はやっぱり動きが早いですね 🚤

**Reply by jay-es** [reply]
ありがとうございます。自分も問題ないと思います！
チェックリスト + 全ページの issue を作っておくと管理しやすそうだなと Vue 3 の翻訳を見て思いました。
textlint や用語集も使いたいですね！

以前どなたかが「外来語は長音記号ありが主流になってきているようだから切り替える？」のような内容をつぶやいているのを見たような気がするのですがいかがしましょうか？（気のせいだったらすみません:bow:）

**Reply by kazupon** [reply]
@naokie 
ありがとうございます！

textlint と用語集、忘れてしました。大事ですね！やりましょう！

@jay-es 

> 以前どなたかが「外来語は長音記号ありが主流になってきているようだから切り替える？」のような内容をつぶやいているのを見たような気がするのですがいかがしましょうか？（気のせいだったらすみません🙇）

ああ、そうですね。そうした方が、例えば、コンピュータかコンピューターなのか悩まなくていいので、そちらの方がよさそうです！

**Reply by kazupon** [reply]
@jay-es 
大方、方針大丈夫そうなので、この repository、まずはセットアップお願いしていいでしょうか？ 🙏 
中国がすでに先行しているので、参考になると思います。
https://github.com/vuejs-translations/docs-zh-cn

権限が足りない等あって、セットアップできないようでしたら、おっしゃってください！

**Reply by jay-es** [reply]
@kazupon 了解しました。
本家のコミット取り込み、textlint や ryu-cho の設定などを進めてみます！

**Reply by jay-es** [reply]
@kazupon 
中国版のリポジトリを確認したところ本家の過去のコミットも取り込まれていたので、それにならい `https://github.com/vuejs/docs` のリポジトリをクローンし、こちらのリポジトリに丸ごと反映させようとしたのですが権限が足りませんでした。

```shell
$ git push --mirror https://github.com/vuejs-translations/docs-ja.git
remote: Permission to vuejs-translations/docs-ja.git denied to jay-es.
fatal: unable to access 'https://github.com/vuejs-translations/docs-ja.git/': The requested URL returned error: 403
```

いくつかのやり方が考えられますが、どのようにするのがよいでしょうか
- jay-es アカウントに権限を付与していただく
- 権限が必要なのはおそらく最初だけなので、このコマンドだけ権限がある方にやっていただく
- https://github.com/vitejs/docs-ja のように現時点の本家の main ブランチの内容を `initial commit` としてコミットする
- その他

**Reply by kazupon** [reply]
@jay-es 
一旦私の方で、`vuejs/docs` から `vuejs-jp/docs` に fork してそれを `vuejs-translations/docs-ja` に push しました。
https://github.com/vuejs-jp/docs

とりあえず、最初の repo はセットアップしたので、よろしくお願いします。 🙏

**Reply by jay-es** [reply]
@kazupon ありがとうございます！ お手間を取らせてしまいすみません:bow:

**Reply by jay-es** [reply]
@kazupon @naokie 一通り準備できましたのでレビュー依頼を出させていただきました。
変更すべき点や足りない点がありましたらご指摘ください。
また、いくつか疑問点も記載してありますので、ご確認いただけると助かります:pray:
https://github.com/vuejs-translations/docs-ja/pull/3

**Reply by kazupon** [reply]
@jay-es ありがとうございます！レビューします！

**Reply by jay-es** [reply]
@kazupon @naokie Guide についての翻訳 issue をすべて作成いたしました https://github.com/vuejs-translations/docs-ja/issues/6

これで全体の約半分なのですが、残りのページも issue を作った方がよろしいでしょうか？

**Reply by kazupon** [reply]
@jay-es 
ありがとうございます!!
そうですね、残りページはよろしくお願いします！

**Reply by jay-es** [reply]
@kazupon @naokie 
残りの issue も作成完了しました。 #84 #100 #114
探しやすいように「翻訳まとめ 」タグをつけました

ryu-cho について。yaml を修正して動き始めましたが、それ以前の差分（本家の 1/30-2/4 のコミット）の PR が作られていないので、手動で取り込んでしまった方がよろしいでしょうか？

**Reply by kazupon** [reply]
@jay-es 
ありがとうございます！
すごいです！本当に助かります！

差分については、そうですね。
もし、できるのでしたら、お願いしたいです！

**Reply by jay-es** [reply]
@kazupon @naokie
ryu-cho 稼働直前の状態を取り込んで PR 作成しました #134
また、ryu-cho が PR を作成しなかった issue についても対応したので承認お願いします:pray: #135 

ryu-cho が自動生成した PR についてはどんどんマージしていってよいですかね？

**Reply by kazupon** [reply]
@jay-es 
ありがとうございます！
上記、2つのPR、approve しました。

はい、jay-es さん、vuejs-translations の ja チームで admin 権限があるはずなので、どんどん翻訳作業やっていっても大丈夫です！

**Reply by naokie** [reply]
翻訳しながらも差分は徐々に発生していくでしょうから、
特に未翻訳ページについての差分はレビューなしで、取り込みしちゃって大丈夫でしょう。

**Reply by jay-es** [reply]
ryu-cho の PR をマージしておきました！

**Reply by jay-es** [reply]
CONTRIBUTING.md の「貢献方法」は ryu-cho が生成した issue に対する内容なので、未翻訳ページの場合はまとめの方にコメントしてもらうように追記した方がよいでしょうか。

また、一定期間（2週間 or ひと月くらい？）連絡なしで進捗がない場合はアサインを外す旨の注意書きもあるとよいかなと思いました（Vite の時に何件かあったので...）

**Reply by kazupon** [reply]
@jay-es 
> CONTRIBUTING.md の「貢献方法」は ryu-cho が生成した issue に対する内容なので、未翻訳ページの場合はまとめの方にコメントしてもらうように追記した方がよいでしょうか。

確かに、そうですね。
未翻訳(新規)ページの翻訳ガイドもあると、どう対応したらいいのかわかりやすいので、載せておくとフレンドリーでいいですね！

> また、一定期間（2週間 or ひと月くらい？）連絡なしで進捗がない場合はアサインを外す旨の注意書きもあるとよいかなと思いました（Vite の時に何件かあったので...）

そうですね。2週間ぐらい進展がない場合は、その旨書いておいた方がいいですね。

**Reply by kazupon** [reply]
@jay-es 
1つ気づいたのですが、node のバージョンを v14.19.0 以上にして、node に同包される corepack を使っておくと pnpm までの導入を簡易化できるので、package.json に `packageManager` フィールドを追加しておくと良さそうです。

詳しくはこちら。
- node.js 公式: https://nodejs.org/api/corepack.html
- zenn corepack 記事: https://zenn.dev/teppeis/articles/2021-05-corepack

**Reply by jay-es** [reply]
textlint にカタカナ長音についてのルールがあるので textlint-rule-preset-vuejs-jp で使えるように PR を出そうと思います。
（文末などでは動かないようですが、ある程度は統一できそうなので）

**Reply by kazupon** [reply]
ありがとうございます！

**Reply by kazupon** [reply]
@jay-es 
こちら早速翻訳して頂いてありがとうございます！
https://github.com/vuejs-translations/docs-ja/pull/206

準備できたようなので、明日月曜、募集をかけようと思いますが、準備の方はもう大丈夫な感じですか？

**Reply by jay-es** [reply]
issue などの作成は一通り終わっていますが、念のため `textlint-rule-preset-vuejs-jp` の準備ができてからがベターかと思います。
（あとからパッケージのバージョンや設定を変えると、参加者への周知やルール適用前にマージした箇所の修正が大変そうなので...:bow:）

**Reply by naokie** [reply]
textlint-rule-preset への変更取り込んでおきました！

**Reply by jay-es** [reply]
ありがとうございます！ 反映して PR 作ります:raising_hand_man:

**Reply by naokie** [reply]
@jay-es 
#237 こちらブロッカーになってましたね。遅くなりました 🙇🏼 
問題なかったのでレビュー&マージしておきました。

@kazupon 
これが最後の準備だったと思うので、募集かけていきましょう！

**Reply by kazupon** [reply]
@naokie 
そうですね。来週月曜に募集開始しましょう！

**Reply by kazupon** [reply]
@jay-es @naokie 
募集を開始するにあたって、翻訳したい方向けの導線となる入り口を兼ねた discussion を作りました。
https://github.com/vuejs-translations/docs-ja/discussions/273

@jay-es 
ちょっと気づいたんですが、各 issue に `翻訳準備中` というラベルがあるのですが、これ外してもらうことできますか？ 🙏 
現状のままだと、ちょっと翻訳していいのかどうか、分からないと思うので。 😅

**Reply by kazupon** [reply]
GitHub issues にあるラベル眺めて思ったんですが、もしかして、`翻訳準備中` は、我々の翻訳に向けて準備をしているステータスのことではなく、実際に翻訳未着手のことを意味してます？

**Reply by jay-es** [reply]
@kazupon 以前、issue に翻訳希望のコメントがついてしまったので、まだ募集していないことを示すためにつけていました。
募集開始時に「翻訳募集中」にリネームする予定でした（CONTRIBUTING.md の `1.` のリンク先で結果が反映されるようになるはず）。
もうリネームしてしまった方がよろしいでしょうか？

また、募集開始後に翻訳希望者が現れてアサインしたら issue からは「翻訳募集中」をはずして「翻訳中」タグをつけていこうと思っていたのですが、この運用で問題ないでしょうか？（事前に確認しようと思って忘れていました:bow:）

**Reply by kazupon** [reply]
@kazupon 
ありがとうございます！
ラベルの件、理解しました！

そうですね。翻訳を始めたいのでリネームしてお願いしたいです！

>CONTRIBUTING.md の 1. のリンク先

そうなると、#273 はシンプルに `CONTRIBUTING.md` に導線があるだけでよい気がしてきました。

>また、募集開始後に翻訳希望者が現れてアサインしたら issue からは「翻訳募集中」をはずして「翻訳中」タグをつけていこうと思っていたのですが、この運用で問題ないでしょうか？

ありがとうございます、その運用で問題ないと思います！

**Reply by kazupon** [reply]
それか、`README.md` をもう少し、翻訳にしやすいような感じの更生 & 導線があるとよさそうです。
https://github.com/vuejs-translations/docs-ja/blob/main/README.md

**Reply by jay-es** [reply]
@kazupon タグのリネーム完了しました。
たしかに `README.md` は原文をほぼそのまま翻訳しただけなので、参加者向けの内容にした方がよさそうですね。
後ほど PR 作成してみます。

**Reply by jay-es** [reply]
PR 作成しました #276
CONTRIBUTING.md への導線を追加し、翻訳と関係ない内容についてはコメントアウトしました

**Reply by kazupon** [reply]
@jay-es PRマージしましたが、明日は休日ですので、24日木曜日にアナウンスしますね。🙂

**Reply by jay-es** [reply]
了解です！ よろしくお願いします:pray:

**Reply by jay-es** [reply]
PR 承認後そのままマージされてしまうかもしれないので、2 人分の承認がないとマージできないよう設定しておきました。
問題ありそうなら戻します！
